frontend/src/framework/StatusWriter.tsx

```diff
@@ -0,0 +1,77 @@
+import React from "react";
+
+import { ModuleContext } from "./ModuleContext";
+import { ModuleInstanceStatusController, StatusMessageType, StatusSource } from "./ModuleInstanceStatusController";
+
+export class ViewStatusWriter {
+    private _statusController: ModuleInstanceStatusController;
+
+    constructor(moduleContext: ModuleContext<any>) {
```

Comment:
Maybe just pass the `ModuleInstanceStatusController` directly in the constructor?
Slightly more direct and slightly more cumbersome for the user. After all, we want them to use the hooks.
Same goes for the `SettingsStatusWriter`